nipype/interfaces/io.py

```diff
@@ -2781,18 +2788,17 @@ def __init__(self, infields=None, **kwargs):
 
         if not isdefined(self.inputs.output_query):
             self.inputs.output_query = {
-                "func": {"modality": "func", 'extensions': ['nii', '.nii.gz']},
-                "anat": {"modality": "anat", 'extensions': ['nii', '.nii.gz']},
+                "bold": {"datatype": "func", "suffix": "bold",
+                         "extensions": ["nii", ".nii.gz"]},
+                "T1w": {"datatype": "anat", "suffix": "T1w",
+                        "extensions": ["nii", ".nii.gz"]},
                 }
 
         # If infields is empty, use all BIDS entities
         if infields is None:
-            # Version resilience
-            try:
-                from bids import layout as bidslayout
-            except ImportError:
-                from bids import grabbids as bidslayout
-            bids_config = join(dirname(bidslayout.__file__), 'config', 'bids.json')
+            from bids import layout as bidslayout
+            bids_config = join(
+                dirname(bidslayout.__file__), 'config', 'bids.json')
             bids_config = json.load(open(bids_config, 'r'))
             infields = [i['name'] for i in bids_config['entities']]
```

Comment:
Can we replace all of this with `BIDSLayout().entities`? I'm not a fan of digging through the directory structure of a package unless there are no other options.

Reply:
Oh, sorry, I meant that in a pseudocode sense of the entities attribute of an instantiated object, not that that snippet would work.

Reply:
Yeah, maybe I can reorganize this to set up the layout first, before setting the output traits. 

Is there any problem with instantiating the layout in `__init__`?

Reply:
The biggest problem is that inputs can change between instantiation and running.

Reply:
That's a pretty big problem! 

So I think for now instantiating a layout just to get entities does not work. Adding a helper function to pybids and then changing this later is a better option.